packages/flutter_tools/lib/src/project.dart

```diff
@@ -229,38 +229,64 @@ class FlutterProject {
     return manifest;
   }
 
+  /// Re-generates project files necessary to make Gradle builds work on Android
+  /// and CocoaPods+Xcode work on iOS, for app and module projects only.
+  ///
+  /// Will not create project platform directories if they do not already exist.
+  Future<void> regeneratePlatformSpecificTooling() async {
+    return ensureReadyForPlatformSpecificTooling(
+      androidPlatform: android.existsSync(),
+      iosPlatform: ios.existsSync(),
+      // TODO(stuartmorgan): Revisit the conditions here once the plans for handling
+      // desktop in existing projects are in place.
```

Comment:
@stuartmorgan I moved the TODOs scattered around to reexamine the logic to this spot.  Let me know if you want it somewhere else, or need it reworded.

Reply:
The TODO was really about figuring out what `checkProjects` was for and why we would ever want the `checkProjects = false` behavior since it seemed really bizarre, so it looks like this PR obsoletes it. But I can clean it up later.